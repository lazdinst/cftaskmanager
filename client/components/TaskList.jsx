import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { deleteTask } from '../store/actions/app';

import TaskListEntry from './TaskListEntry.jsx';

import { Container, Card, Button } from 'semantic-ui-react'

function TaskList({tasks, deleteTask}) {
  tasks.sort(function(a,b) { return new Date(b.createdAt) - new Date(a.createdAt); });
  const taskEntries = tasks.map((task,i) => (<TaskListEntry task={task} taskNum={i+1} key={i} />));
  return (
    <Container text>
      <Card.Group>
        {taskEntries || []}
      </Card.Group>
    </Container>
  );
}

const mapStateToProps = (state) => ({
  tasks: state.app.tasks,
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  deleteTask
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TaskList);