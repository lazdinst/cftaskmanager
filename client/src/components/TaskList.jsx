import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import TaskListEntry from './TaskListEntry.jsx';
import { Container, Card } from 'semantic-ui-react'

const TaskList = ({tasks}) => {
  let taskEntries = tasks.map((task,i) => { return <TaskListEntry task={task} key={i} />; });
  return (
    <Container text>
      <Card.Group>
        {taskEntries || []}
      </Card.Group>
    </Container>
  );
}

const mapStateToProps = (state) => ({
  tasks: state.app.tasks,
});

export default connect(
  mapStateToProps
)(TaskList);