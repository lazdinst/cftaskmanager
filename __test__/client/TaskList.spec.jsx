import React from 'react'
import { shallow, mount } from 'enzyme';
import TaskList from '../../client/components/TaskList.jsx'
import toJson from 'enzyme-to-json';

xdescribe('<TaskList />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TaskList />);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
