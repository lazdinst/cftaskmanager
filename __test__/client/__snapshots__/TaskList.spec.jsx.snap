// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskList /> should render 1`] = `undefined`;
